### DISCUSSION TITLE
how can set header in axios config with out error Refused to set unsafe header "origin"?

### DISCUSSION THREAD

**Discussion by javad-sh** [vercel/next.js#50422] [Help]
### Summary

hey friends.
I used cors in back-end so I have to set origin in headers of requests that send to serevr.
To do this is set origin in my axios options. 
Every things work but this error comes in browser console Refused to set unsafe header "origin"?
how can I fix that ?
how can I set headers for my requests with out this error ?

**Comment by devjiwonchoi** [comment]
When you have already set the `CORS origin` option on the backend based on your frontend's URL, you don't need an extra CORS option on the Request header since it is generally recommended to set the CORS option on the `Response`, which means to be set on the API server.

The reason why you have `Refused to set the unsafe header "origin"` is that you attempted to set the CORS option on the `Request` header which is not recommended for security reasons since it is recommended to control the CORS on the backend.

In summary, it is OK to remove your Axios CORS option if you have already allowed the cross-origin for your frontend to the backend.
